crates/starknet-replay/src/rpc_state_reader.rs

```diff
@@ -0,0 +1,144 @@
+use blockifier::execution::contract_class::ContractClass;
+use blockifier::state::errors::StateError;
+use blockifier::state::state_api::{StateReader, StateResult};
+use starknet::core::types::BlockId;
+use starknet::providers::jsonrpc::JsonRpcTransport;
+use starknet::providers::{JsonRpcClient, Provider, ProviderError};
+use starknet_api::core::{ClassHash, CompiledClassHash, ContractAddress, Nonce};
+use starknet_api::hash::StarkFelt;
+use starknet_api::state::StorageKey;
+use starknet_os_types::deprecated_compiled_class::GenericDeprecatedCompiledClass;
+use starknet_os_types::hash::GenericClassHash;
+use starknet_os_types::sierra_contract_class::GenericSierraContractClass;
+
+use crate::utils::{execute_coroutine, felt_api2vm, felt_vm2api};
+
+pub struct AsyncRpcStateReader<T>
```

Comment:
Is `T` here just for the `JsonRpcClient<T>`? We could give it the concrete type here of `HttpTransport` to make our lives easier unless we have a good reason to keep that generic.

Reply:
Meh, this wasn't a big deal, I wouldn't change it unless it makes sense